mlflow/projects/__init__.py

```diff
@@ -671,10 +700,41 @@ def _validate_docker_installation():
                                  "at https://docs.docker.com/install/overview/.")
 
 
-def _validate_docker_env(docker_env):
-    if not docker_env.get('image'):
+def _validate_docker_env(project):
+    if not project.name:
+        raise ExecutionException("Project name in MLProject must be specified when using docker "
+                                 "for image tagging.")
+    if not project.docker_env.get('image'):
         raise ExecutionException("Project with docker environment must specify the docker image "
-                                 "to use via an 'image' field under the 'docker_env' field")
+                                 "to use via an 'image' field under the 'docker_env' field.")
+
+
+def _parse_kubernetes_config(backend_config, work_dir):
+    """
+    Creates build context tarfile containing Dockerfile and project code, returning path to tarfile
+    """
+    kube_config = backend_config.copy()
+    if not backend_config:
+        raise ExecutionException("Backend_config file not found.")
+    if 'kube-job-template-path' in backend_config.keys():
+        kube_job_template = backend_config['kube-job-template-path']
+        if os.path.exists(os.path.join(work_dir, kube_job_template)):
```

Comment:
I don't think the `kube-job-template-path` should be relative to `MLproject` since it may contain user specific configurations like namespace.

Reply:
`os.path.join` just joins the two paths only if the second is relative (without a `/` in the beginning), so if you pass the absolute path it won't join with the mlflow work_dir.

You can check here in [docs](https://docs.python.org/2/library/os.path.html#os.path.join):
> If a component is an absolute path, all previous components are thrown away and joining continues from the absolute path component.

I think we can let as it is so both use cases, absolute and relative paths, are valid. What do you think?

Reply:
Since the `backend_config` is loaded relative to the CWD, I think it only makes sense that the `kube-job-template-path` is also loaded relative to the CWD. We can discuss more offline if you'd like!

Reply:
I've just checked it here, the loading of  `backend_config` also behave this way, We can pass an absolute or relative path and the `open` function will handle the same way `os.path.join` handles it. You can see the [docs](https://docs.python.org/3/library/functions.html#open) here.
> file is a path-like object giving the pathname (absolute or relative to the current working directory) 

We can still chat offline if you'd like too.